crates/ruma-common/src/events/call/answer.rs

```diff
@@ -18,16 +25,80 @@ pub struct CallAnswerEventContent {
     /// The VoIP session description object.
     pub answer: AnswerSessionDescription,
 
-    /// The ID of the call this event relates to.
+    #[cfg(not(feature = "unstable-msc2746"))]
+    /// A unique identifier for the call.
+    ///
+    /// With the `unstable-msc2746` feature, this uses the stricter `VoipId` type.
     pub call_id: String,
 
+    #[cfg(feature = "unstable-msc2746")]
+    /// A unique identifier for the call.
+    ///
+    /// Without the `unstable-msc2746` feature, this can be any string.
+    pub call_id: OwnedVoipId,
+
+    #[cfg(feature = "unstable-msc2746")]
+    /// **Required in version 1.** A unique ID for this session for the duration of the call.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub party_id: Option<OwnedVoipId>,
+
+    #[cfg(not(feature = "unstable-msc2746"))]
     /// The version of the VoIP specification this messages adheres to.
+    ///
+    /// With the `unstable-msc2746` feature, this can be a `UInt` or a `String`.
```

Comment:
Hrm, that makes this feature not additive. The alternative would be to have `CallVersion` unconditionally, and just `CallVersion::Namespaced` feature-gated. Doesn't seem like a great solution either, but may be better than a non-additive feature… wdyt?

Reply:
I forgot to add a comment here, it seems I misunderstood the definition of the version. The MSC in its current state actually says that we can receive either `0` as a number, or a string. I started a conversation there to say that it's not backwards-compatible though.

There's also the fact that Element Web sends `1` as a number while Element Android sends it as a string.

I think `CallVersion` should now be an enum with a list of known versions and custom serde.

And finally to answer your question, I also think that it's better to make it additive.

There's also the `call_id` field that is non-additive, no?

Reply:
I missed that there's another one. That one looks to be a breaking change? What's the story behind that, do all current implementations conform to the stricter requirements anyways?

Reply:
It seems like it but I don't know for sure. I [left a comment](https://github.com/matrix-org/matrix-spec-proposals/pull/2746/files#r883701785).

Reply:
For now it could just accept any string but provide a constructor that respects this MSC

Reply:
So I made `VoIP` a "semantic" type like `TransactionId` for now, And I changed `CallVersion` to `VoipVersionId` to be closer to the spec, and it's now an enum with known versions.